### PR TITLE
feat(api-gateway): support v2 http api

### PR DESCRIPTION
**Pull request by ophiryael**
**REST:**
Screen Shot 2021-09-12 at 18 07 10

**HTTP:**
Screen Shot 2021-09-12 at 18 07 28

### CODE REVIEW

File: epsagon/lambda_trigger.go [hunk 1]
```diff
@@ -43,50 +35,106 @@ func mapParametersToString(params map[string]string) string {
...
-				Time:      tracer.GetTimestamp(),
-			})
-			triggerEvent.Resource.Metadata["body"] = ""
-		} else {
-			triggerEvent.Resource.Metadata["body"] = string(bodyJSON)
-		}
-		triggerEvent.Resource.Metadata["headers"] = mapParametersToString(event.Headers)
+}
+
+func addAPIGatewayEventMetadata(triggerEvent *protocol.Event, eventFields *APIGatewayEventFields) {
```
Comment (maorlx) [inline epsagon/lambda_trigger.go:85]: would rename the func name addXXXXPayload/non metadata (as its being called if !metadataOnly)

File: epsagon/lambda_trigger.go [hunk 2]
```diff
@@ -43,50 +35,106 @@ func mapParametersToString(params map[string]string) string {
...
+		triggerEvent.Resource.Metadata["body"] = string(bodyJSON)
 	}
+	triggerEvent.Resource.Metadata["headers"] = mapParametersToString(eventFields.headers)
+}
 
-	return triggerEvent
+func triggerAPIGatewayProxyRequest(rawEvent interface{}, metadataOnly bool) *protocol.Event {
+	event, ok := rawEvent.(lambdaEvents.APIGatewayProxyRequest)
+	if !ok {
+		assertionException := getEventAssertionException(rawEvent, "lambdaEvents.APIGatewayProxyRequest")
```
Comment (maorlx) [inline epsagon/lambda_trigger.go:103]: is there a check for the returned event value? (to make sure the code won't panic when the func returns nil)

File: epsagon/lambda_trigger.go (hunk 1 above)
Comment (ophiryael) [reply epsagon/lambda_trigger.go:85]: Good idea, done.

File: epsagon/lambda_trigger.go (hunk 2 above)
Comment (ophiryael) [reply epsagon/lambda_trigger.go:103]: Yes, the `addLambdaTrigger` function that initiates the trigger event creation flow ignores `nil` events.
(line 495, lambda_trigger.go)
```go
func addLambdaTrigger(
	payload json.RawMessage,
	metadataOnly bool,
	triggerFactories map[string]factoryAndType,
	currentTracer tracer.Tracer,
) {
	var triggerEvent *protocol.Event

	triggerSource := guessTriggerSource(payload)

	if triggerSource == "json" {
		triggerEvent = triggerJSONEvent(payload, metadataOnly)
	} else if triggerSource == "api_gateway_no_proxy" {
		// currently not supported, needs to extract data from json
	} else {
		factoryStruct, found := triggerFactories[triggerSource]
		if found {
			triggerEvent = decodeAndUnpackEvent(
				payload, factoryStruct.EventType, factoryStruct.Factory, metadataOnly)
		}
	}

	// If a trigger was found
	if triggerEvent != nil {
		currentTracer.AddEvent(triggerEvent)
	}
}

```

**Review by maorlx — APPROVED** [review]
looks good :D

**Pull request merged**